### PR TITLE
Fix bug 286 random token replacement

### PR DESCRIPTION
**Pull request by li-wu**
Replace with `random.SystemRandom`.

### CODE REVIEW

**Comment by szhong26** [comment]
It looks like this is still not correct. Here is a sample output for the sample CPUTime.perfmon
[eventgen_data_output.txt](https://github.com/splunk/eventgen/files/3502456/eventgen_data_output.txt)

File: splunk_eventgen/lib/eventgentoken.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def _getReplacement(self, old=None, earliestTime=None, latestTime=None, s=None,
                 except:
                     logger.error("Could not parse json for '%s' in sample '%s'" % (listMatch.group(1), s.name))
                     return old
-                return random.choice(value)
+                return random.SystemRandom().choice(value)
```
Comment (li-wu) [inline splunk_eventgen/lib/eventgentoken.py]: Use `SystemRandom` class which uses the system function `os.urandom()` to generate random numbers from sources provided by the operating system.

File: splunk_eventgen/lib/plugins/generator/perdayvolumegenerator.py [hunk 2]
```diff
@@ -31,17 +31,15 @@ def gen(self, count, earliest, latest, samplename=None):
 
         # Create a counter for the current byte size of the read in samples
         currentSize = 0
-        # Replace event tokens before calculating the size of the event
-        updated_sample_dict = GeneratorPlugin.replace_tokens(self, self._sample.sampleDict, earliest, latest)
```
Comment (li-wu) [inline splunk_eventgen/lib/plugins/generator/perdayvolumegenerator.py]: Use the raw sample to calculate the size of event and replace the token later in `build_events`.

File: splunk_eventgen/lib/plugins/generator/perdayvolumegenerator.py [hunk 3]
```diff
@@ -63,28 +61,28 @@ def gen(self, count, earliest, latest, samplename=None):
...
                 sizeremaining = size - currentreadsize
-                targetlinesize = len(updated_sample_dict[targetline]['_raw'])
+                targetlinesize = len(self._sample.sampleDict[targetline]['_raw'])
                 if size < targetlinesize:
                     logger.error(
                         "Size is too small for sample {}. We need {} bytes but size of one event is {} bytes.".format(
                             self._sample.name, size, targetlinesize))
                     break
-                if targetlinesize <= sizeremaining or targetlinesize * .9 <= sizeremaining:
+                if targetlinesize <= sizeremaining:
```
Comment (li-wu) [inline splunk_eventgen/lib/plugins/generator/perdayvolumegenerator.py]: Remove redundant logic here.

**Comment by li-wu** [comment]
Per feedback from @szhong26 ,  this pr now can get random value replacement for token.

**Pull request merged**